cachito/workers/pkg_managers/gomod.py

```diff
@@ -198,7 +198,10 @@ def resolve_gomod(app_source_path, request, dep_replacements=None, git_dir_path=
         else:
             log.info("Downloading the gomod dependencies")
             run_download_cmd(("go", "mod", "download"), run_params)
-        if dep_replacements:
+        tidy_compat = "1.17"
```

Comment:
> The ticket says, Add a tidy_compat configuration option, that, when set, controls the value of the -compat flag passed to the go mod tidy invocation. I've implemented tidy_compat temporarily as a local variable, but how should this new variable be passed? As an additional input parameter to resolve_gomod?

There are 2 options: 
1. use a configuration inside:
https://github.com/containerbuildsystem/cachito/blob/master/cachito/workers/config.py
2. use flags in a request

I would like to see the second option, so it will be up to our users to set. 
@brunoapimentel @serg-cymbaluk @ejegrova WDYT?

Reply:
I believe a mix of both options would work well. We should have a default value for this variable even if the user doesn't explicitly provide the value.

Reply:
The default being the newest version of Go, of course.

Reply:
Upon reviewing, we shouldn't store a default value in `config.py`. The default behavior without an explicit `tidy_compat` value should be a simple `go mod tidy` command without `-compat=`.

Reply:
It should be a new optional flag to be used during the creation of the request. That way, the user will have control on how/if it's used.